src/legacy/core_plugins/data/public/filter/filter_manager/filter_state_manager.ts

```diff
@@ -21,6 +21,13 @@ import _ from 'lodash';
 import { State } from 'ui/state_management/state';
 import { FilterManager, esFilters } from '../../../../../../plugins/data/public';
 
+import {
+  compareFilters,
```

Comment:
nit: didn't understand why we don't want to expose it publicly and consume from esFilters.compareFilters?

Reply:
I see no reason ATM to add this function to the public API.
Instead of exporting it, then deleting it once you finish handling state,
I just prefer to import it from an internal path, until this is deleted.

Reply:
ok, but seems like when I finish, likely still will need to be exported, as state management stuff is outside `/data` plugin.

Reply:
I think the filter state management part should be contained inside `filter_manager` \ elsewhere in the data plugin.
I wouldn't want each app syncing the state back and forth for filters \ timefilter \ query.